object/proxy.go

```diff
@@ -121,7 +121,11 @@ func (p *Proxy) SetAttr(name string, value Object) error {
 			return err
 		}
 		if field.CanSet() {
-			field.Set(reflect.ValueOf(result))
+			if result == nil {
+				field.SetZero()
```

Comment:
not sure if in case where `field.Kind() == reflect.Ptr` and `result == nil` we should call `fielld.SetZero()`, souldn't we set the field to `nil`?

Reply:
I'm going to merge this PR since it's been lingering but we should still look closer at this question.